internal/kubernetes/operator/deployment/deployment.go

```diff
@@ -246,14 +248,14 @@ func buildBackups(backupsProvider store.ScheduleDescriber, projectID, clusterNam
 			APIVersion: "atlas.mongodb.com/v1",
 		},
 		ObjectMeta: v1.ObjectMeta{
-			Name:      strings.ToLower(fmt.Sprintf("%s-backupschedule", clusterName)),
+			Name:      resources.NormalizeAtlasResourceName(fmt.Sprintf("%s-backupschedule", clusterName)),
```

Comment:
empirically I feel this reads better and will be better long term 
```suggestion
			Name:      fmt.Sprintf("%s-backupschedule", resources.NormalizeAtlasResourceName(clusterName)),
```

But since this is empirical don't take it as a blocker